d2/src/test/java/com/linkedin/d2/balancer/dualread/UriPropertiesDualReadMonitorTest.java

```diff
@@ -114,6 +124,75 @@ public void testReportData() {
     verifyJmxMetricParams(fixture, CLUSTER_2, null, 1.0);
   }
 
+  @DataProvider
+  public Object[][] reportDataInMultiThreadsDataProvider() {
+    // simulate different orders that the data is reported between the old and new Lbs.
+    return new Object[][] {
+        {
+            Arrays.asList(
+                new ImmutablePair<>(URI_PROPERTIES_1, false),
+                new ImmutablePair<>(URI_PROPERTIES_1, true),
+                new ImmutablePair<>(URI_PROPERTIES_URI_1_AND_2, false),
+                new ImmutablePair<>(URI_PROPERTIES_URI_1_AND_2, true)
+            )
+        },
+        {
+          Arrays.asList(
+            new ImmutablePair<>(URI_PROPERTIES_1, false),
+            new ImmutablePair<>(URI_PROPERTIES_URI_1_AND_2, false),
+            new ImmutablePair<>(URI_PROPERTIES_1, true),
+            new ImmutablePair<>(URI_PROPERTIES_URI_1_AND_2, true)
+            )
+        },
+        {
+            Arrays.asList(
+                new ImmutablePair<>(URI_PROPERTIES_1, false),
+                new ImmutablePair<>(URI_PROPERTIES_1, true),
+                new ImmutablePair<>(URI_PROPERTIES_URI_1_AND_2, true),
+            new ImmutablePair<>(URI_PROPERTIES_URI_1_AND_2, false)
+            )
+        },
+        {
```

Comment:
Inconsistent formatting?

Reply:
yeah, forgot to hit the auto-formatting before submission. Updated.